internal/vm/opcodes/proposal_update_body.go

```diff
@@ -0,0 +1,28 @@
+package opcodes
+
+import (
+	"errors"
+
+	"github.com/git-town/git-town/v21/internal/forge/forgedomain"
+	"github.com/git-town/git-town/v21/internal/messages"
+	"github.com/git-town/git-town/v21/internal/vm/shared"
+)
+
+type ProposalUpdateBody struct {
+	Proposal                forgedomain.ProposalInterface
```

Comment:
Please use `forgedomain.Proposal` here for proper serialization to and from JSON.

```suggestion
	Proposal                forgedomain.Proposal
```